### PR TITLE
Row should be public 

### PR DESCRIPTION
**Pull request by dushibaiyu**

### CODE REVIEW

**Comment by IrenejMarc** [comment]
@dushibaiyu This does not seem correct. Converting bytes to a string and then to a long should only ever be required if the results are textual (e.g. fetched with `exec`, not `execParams`), which `Value` does not (yet) support.

The public visibility of row is another thing, but I see no downsides to it.

**Comment by dushibaiyu** [comment]
Sorry,I don't test execParams.
But Row should be public , when I user the Result.opIndex, and Result.opopApply

**Comment by IrenejMarc** [comment]
Seeing as Row is just a proxy object and should never really be instantianted manually, I decided to mark it `package` visibility. You can still use it with foreach (`foreach (r; result)`), or with opIndex (`auto row = result[0];`).

I see no real reason to make it public, and at this point, I suppose, it is a matter of preference. I'll mark this issue as "Need input" and will merge it later if nobody has anything against it.

**Comment by dushibaiyu** [comment]
auto is good. But I perfer can write: Row row = result[0] ;

File: source/dpq/result.d [hunk 1]
```diff
@@ -247,7 +247,8 @@ struct Result
 	}
 }
 
-package struct Row
+//package
```
Comment (IrenejMarc) [inline source/dpq/result.d]: Remove the comment and I'll merge this. I agree, Row should be public.

Comment (donglei) [inline source/dpq/result.d]: UFCS function is ok for Row

**Pull request merged**